kyo-prelude/shared/src/main/scala/kyo2/Result.scala

```diff
@@ -10,29 +10,30 @@ opaque type Result[+E, +A] >: (Success[A] | Failure[E]) = Success[A] | Failure[E
 object Result:
     import internal.*
 
-    given [E, A](using CanEqual[A, A]): CanEqual[Result[E, A], Result[E, A]] = CanEqual.derived
-    given [E, A]: CanEqual[Result[E, A], Panic]                              = CanEqual.derived
+    inline given [E, A](using inline ce: CanEqual[A, A]): CanEqual[Result[E, A], Result[E, A]] = CanEqual.derived
+    given [E, A]: CanEqual[Result[E, A], Panic]                                                = CanEqual.derived
 
-    def apply[A](expr: => A): Result[Nothing, A] =
+    inline def apply[A](expr: => A): Result[Nothing, A] =
         try
             Success(expr)
         catch
-            case ex: Throwable if NonFatal(ex) => Panic(ex)
+            case ex => Panic(ex)
```

Comment:
The `NonFatal` check is now done by `Panic.apply`. This reduces the size of the inlined bytecode.